src/sha2.rs

```diff
@@ -1,27 +1,34 @@
-#[cfg(not(openssl))]
-pub use sha2::{Digest, Sha256};
-
-#[cfg(openssl)]
+#[cfg(feature = "openssl")]
 use openssl;
 
-#[cfg(openssl)]
-#[derive(Clone)]
+#[cfg(not(feature = "openssl"))]
+use sha2::Digest;
+
+#[derive(Default, Clone)]
```

Comment:
is `Clone` important? It makes it easier to accidentally copy the context, where it's meant to be moved or passed by reference

Reply:
I think Brandon told me to put that I'll ask him

Reply:
`Clone` was already there before this PR - I just added `Default` and changed the order.

`sha2` and `openssl` already implement `Clone` (which is why it can be derived).

Reply:
i was wrong it was already there, i can remove it if you wish